libtenzir/builtins/operators/show.cpp

```diff
@@ -0,0 +1,451 @@
+              (void)err;
+            },
+            [&](const enumeration_type& e) {
+              auto enum_field = structure.push_field("enum");
+              auto list = enum_field.push_list();
+              for (auto field : e.fields()) {
+                auto field_record = list.push_record();
+                auto name = field_record.push_field("name");
+                auto err = name.add(field.name);
+                auto key = field_record.push_field("key");
+                err = key.add(uint64_t{field.key});
+                (void)err;
+              }
+            },
+            [&](const list_type& l) {
+              auto list = structure.push_field("list");
+              auto err = list.add(fmt::to_string(l.value_type()));
+              (void)err;
+            },
+            [&](const map_type& m) {
+              auto map = structure.push_field("map");
+              auto record = map.push_record();
+              auto key_field = record.push_field("key");
+              auto err = key_field.add(fmt::to_string(m.key_type()));
+              auto value_field = record.push_field("value");
+              err = value_field.add(fmt::to_string(m.value_type()));
+              (void)err;
+            },
+            [&](const record_type& r) {
+              auto record = structure.push_field("record");
+              auto list = record.push_list();
+              for (const auto& field : r.fields()) {
+                auto field_record = list.push_record();
+                auto field_name = field_record.push_field("name");
+                auto err = field_name.add(field.name);
+                auto field_type = field_record.push_field("type");
+                err = field_type.add(fmt::to_string(field.type));
+                (void)err;
+              }
+            },
+          };
+          caf::visit(f, type);
+          auto attributes = collect(type.attributes());
+          if (attributes.empty())
+            continue;
+          auto list = row.push_field("attributes").push_list();
+          for (auto& attribute : attributes) {
+            auto record = list.push_record();
+            err = record.push_field("key").add(attribute.key);
+            err = record.push_field("value").add(attribute.value);
+          }
+          (void)err;
+        }
+        co_yield builder.finish();
+      } else {
+        die("invalid catalog interaction");
+      }
+    } else {
+      die("unchecked aspect");
+    }
+  }
+
+  auto name() const -> std::string override {
+    return "show";
+  }
+
+  auto detached() const -> bool override {
+    return true;
+  }
+
+  auto location() const -> operator_location override {
+    return operator_location::local;
+  }
+
+  auto optimize(expression const& filter, event_order order) const
+    -> optimize_result override {
+    (void)order;
+    (void)filter;
+    return optimize_result{std::nullopt, event_order::unordered, copy()};
```

Comment:
```suggestion
    return do_not_optimize(*this);
```

Yours might also work, but (strictly speaking) it breaks the contract of `optimize`.